hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/MapTableUtils.java

```diff
@@ -54,4 +73,179 @@ public static long estimatePartitionedMapRowCount(NodeEngine nodeEngine, MapServ
 
         return entryCount * memberCount;
     }
+
+    public static List<MapTableIndex> getPartitionedMapIndexes(
+        NodeEngine nodeEngine,
+        MapContainer mapContainer,
+        List<TableField> fields
+    ) {
+        // Do no return non-concurrent indexes for HD maps.
+        boolean hd = mapContainer.getMapConfig().getInMemoryFormat() == InMemoryFormat.NATIVE;
+        boolean globalIndexEnabled = nodeEngine.getProperties().getBoolean(GLOBAL_HD_INDEX_ENABLED);
+
+        if (hd && !globalIndexEnabled) {
+            return Collections.emptyList();
+        }
+
+        Map<QueryPath, Integer> pathToOrdinalMap = mapPathsToOrdinals(fields);
+
+        List<Index> indexes = mapContainer.getIndexList();
+
+        if (indexes.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        List<MapTableIndex> res = new ArrayList<>(indexes.size());
+
+        for (Index index : indexes) {
+            IndexConfig indexConfig = index.getConfig();
+
+            List<QueryDataType> resolvedFieldConverterTypes = indexConverterToSqlTypes(index.getConverter());
+
+            List<Integer> indexFieldOrdinals = new ArrayList<>(indexConfig.getAttributes().size());
+            List<QueryDataType> indexFieldConverterTypes = new ArrayList<>(indexConfig.getAttributes().size());
+
+            for (int i = 0; i < indexConfig.getAttributes().size(); i++) {
+                String attribute = indexConfig.getAttributes().get(i);
+
+                QueryPath attributePath = QueryPath.create(attribute);
+
+                Integer ordinal = pathToOrdinalMap.get(attributePath);
+
+                if (ordinal == null) {
+                    // No mapping for the field. Stop.
+                    break;
+                }
+
+                if (i >= resolvedFieldConverterTypes.size()) {
+                    // No more resolved converters. Stop.
+                    break;
+                }
+
+                QueryDataType fieldType = fields.get(ordinal).getType();
+                QueryDataType converterType = resolvedFieldConverterTypes.get(i);
+
+                if (!isCompatibleForIndexRequest(fieldType, converterType)) {
+                    // Field and converter types are not compatible (e.g. INT vs VARCHAR).
+                    break;
```

Comment:
It might make sense to log such indexes to avoid situations when an index is not used for no apparent reason.

Reply:
It makes sense to log it somehow. But we need some sort of throttling to avoid log pollution. I would do that in a separate PR.